edocumentation-service/src/main/java/org/dcsa/edocumentation/service/EquipmentService.java

```diff
@@ -0,0 +1,63 @@
+package org.dcsa.edocumentation.service;
+
+import lombok.RequiredArgsConstructor;
+import org.dcsa.edocumentation.domain.persistence.entity.Equipment;
+import org.dcsa.edocumentation.domain.persistence.repository.EquipmentRepository;
+import org.dcsa.edocumentation.service.mapping.EquipmentMapper;
+import org.dcsa.edocumentation.service.util.EnsureResolvable;
+import org.dcsa.edocumentation.transferobjects.EquipmentTO;
+import org.dcsa.skernel.errors.exceptions.ConcreteRequestErrorMessageException;
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.ExampleMatcher;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class EquipmentService extends EnsureResolvable<EquipmentTO, Equipment> {
+  private final EquipmentRepository equipmentRepository;
+  private final EquipmentMapper equipmentMapper;
+
+  private final ExampleMatcher exampleMatcher =
+      ExampleMatcher.matchingAll().withIncludeNullValues().withIgnorePaths("equipmentReference");
```

Comment:
Isn't equipmentReference exactly what you want to look for, but you ignore it instead?

Reply:
after this merge we will look into consolidating this with your PR's equipment service and resolve it that way